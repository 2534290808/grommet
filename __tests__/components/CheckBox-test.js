// Checkbox-test.js
import React from 'react';
import Checkbox from '../../src/js/components/Checkbox';
import renderer from 'react/lib/ReactTestRenderer';

describe('Checkbox', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Checkbox id="test" label="Test Checkbox"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct checked=true rendering', () => {
    const component = renderer.create(
      <Checkbox id="test" checked={true} label="Test Checkbox"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct disabled=true rendering', () => {
    const component = renderer.create(
      <Checkbox id="test" disabled={true} label="Test Checkbox"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct reverse=true rendering', () => {
    const component = renderer.create(
      <Checkbox id="test" reverse={true} label="Test Checkbox"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct toggle=true rendering', () => {
    const component = renderer.create(
      <Checkbox id="test" toggle={true} label="Test Checkbox"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});