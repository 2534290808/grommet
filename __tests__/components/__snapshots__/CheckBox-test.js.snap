exports[`Checkbox has correct checked=true rendering 1`] = `
<label
  aria-label="Test Checkbox"
  className="grommetux-check-box">
  <span>
    <input
      checked={true}
      className="grommetux-check-box__input"
      defaultChecked={undefined}
      disabled={undefined}
      id="test"
      name={undefined}
      onChange={undefined}
      tabIndex="0"
      type="checkbox" />
    <span
      className="grommetux-check-box__control">
      <svg
        className="grommetux-check-box__control-check"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24">
        <path
          d="M6,11.3 L10.3,16 L18,6.2"
          fill="none" />
      </svg>
    </span>
  </span>
  <span
    className="grommetux-check-box__label">
    Test Checkbox
  </span>
</label>
`;

exports[`Checkbox has correct default options 1`] = `
<label
  aria-label="Test Checkbox"
  className="grommetux-check-box">
  <span>
    <input
      checked={undefined}
      className="grommetux-check-box__input"
      defaultChecked={undefined}
      disabled={undefined}
      id="test"
      name={undefined}
      onChange={undefined}
      tabIndex="0"
      type="checkbox" />
    <span
      className="grommetux-check-box__control">
      <svg
        className="grommetux-check-box__control-check"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24">
        <path
          d="M6,11.3 L10.3,16 L18,6.2"
          fill="none" />
      </svg>
    </span>
  </span>
  <span
    className="grommetux-check-box__label">
    Test Checkbox
  </span>
</label>
`;

exports[`Checkbox has correct disabled=true rendering 1`] = `
<label
  aria-label="Test Checkbox"
  className="grommetux-check-box grommetux-check-box--disabled">
  <span>
    <input
      checked={undefined}
      className="grommetux-check-box__input"
      defaultChecked={undefined}
      disabled={true}
      id="test"
      name={undefined}
      onChange={undefined}
      tabIndex="0"
      type="checkbox" />
    <span
      className="grommetux-check-box__control">
      <svg
        className="grommetux-check-box__control-check"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24">
        <path
          d="M6,11.3 L10.3,16 L18,6.2"
          fill="none" />
      </svg>
    </span>
  </span>
  <span
    className="grommetux-check-box__label">
    Test Checkbox
  </span>
</label>
`;

exports[`Checkbox has correct reverse=true rendering 1`] = `
<label
  aria-label="Test Checkbox"
  className="grommetux-check-box">
  <span
    className="grommetux-check-box__label">
    Test Checkbox
  </span>
  <span>
    <input
      checked={undefined}
      className="grommetux-check-box__input"
      defaultChecked={undefined}
      disabled={undefined}
      id="test"
      name={undefined}
      onChange={undefined}
      tabIndex="0"
      type="checkbox" />
    <span
      className="grommetux-check-box__control">
      <svg
        className="grommetux-check-box__control-check"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24">
        <path
          d="M6,11.3 L10.3,16 L18,6.2"
          fill="none" />
      </svg>
    </span>
  </span>
</label>
`;

exports[`Checkbox has correct toggle=true rendering 1`] = `
<label
  aria-label="Test Checkbox"
  className="grommetux-check-box grommetux-check-box--toggle">
  <span>
    <input
      checked={undefined}
      className="grommetux-check-box__input"
      defaultChecked={undefined}
      disabled={undefined}
      id="test"
      name={undefined}
      onChange={undefined}
      tabIndex="0"
      type="checkbox" />
    <span
      className="grommetux-check-box__control">
      <svg
        className="grommetux-check-box__control-check"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24">
        <path
          d="M6,11.3 L10.3,16 L18,6.2"
          fill="none" />
      </svg>
    </span>
  </span>
  <span
    className="grommetux-check-box__label">
    Test Checkbox
  </span>
</label>
`;
